docs/source/migrating.rst

```diff
@@ -0,0 +1,189 @@
+
+Configuration files
+^^^^^^^^^^^^^^^^^^^
+
+-  ``~/.ipython/profile_default/static/custom`` → ``~/.jupyter/custom``
+-  ``~/.ipython/profile_default/ipython_notebook_config.py`` →
+   ``~/.jupyter/jupyter_notebook_config.py``
+-  ``~/.ipython/profile_default/ipython_nbconvert_config.py`` →
+   ``~/.jupyter/jupyter_nbconvert_config.py``
+-  ``~/.ipython/profile_default/ipython_qtconsole_config.py`` →
+   ``~/.jupyter/jupyter_qtconsole_config.py``
+-  ``~/.ipython/profile_default/ipython_console_config.py`` →
+   ``~/.jupyter/jupyter_console_config.py``
+
+``JUPYTER_CONFIG_DIR`` env can be used if ``~/.jupyter`` is not desired.
+
+Data files
+^^^^^^^^^^
+
+Data files are things that are installed, but not configuration. This
+includes kernelspecs and notebook extensions. Data files use
+platform-appropriate locations:
+
+-  OS X: ``~/Library/Jupyter``
+-  Windows: the ``%APPDATA%`` environment variable is used
+-  Elsewhere, ``$XDG_DATA_HOME`` is respected, with the default of
+   ``~/.local/share/jupyter``
+
+In all cases, ``JUPYTER_DATA_DIR`` can be used to specify this location
+explicitly. Data files installed system-wide (e.g. in
+``/usr/local/share/jupyter``) have not changed. Per-user installation of
+these has changed from going in .ipython to the above locations:
+
+Profiles
+~~~~~~~~
+
+IPython has the concept of **profiles**, which are collections of
+configuration and runtime files. Inside the IPython directory
+(``~/.ipython``), there are directories with names like
+``profile_default`` or ``profile_demo``. In each of these are
+configuration files (``ipython_config.py``,
+``ipython_notebook_config.py``) and runtime files (``history.sqlite``,
+``security/kernel-*.json``). Profiles could be used to switch between
+configurations of IPython. **Jupyter does not have profiles**.
+
+People could use commands like ``ipython notebook --profile demo`` to
+set the profile for *both* the notebook server and the IPython kernel.
+This is no longer possible in one go with Jupyter, just like it wasn't
+possible in IPython 3 for any other kernels. If you wanted to change the
+notebook configuration, you can set the ``JUPYTER_CONFIG_DIR``:
+
+::
+
+    JUPYTER_CONFIG_DIR=./jupyter_config jupyter notebook
+
+If you just want to change the config file, you can do:
+
+::
+
+    jupyter notebook --config=/path/to/myconfig.py
+
+If, instead, you do want to change the IPython kernel's profile, you
+can't do this at the server command-line anymore. Kernel arguments must
+be done by modifying the kernelspec. The good thing about this is that
+you can do it without relaunching the server. Kernelspec changes take
+effect every time you start a new kernel. The less good thing is that
+there isn't a great way to modify the kernelspecs. You can see where the
+file is with ``jupyter kernelspec list``, and then modify
+``kernels/python3/kernel.json`` by hand.
+
+Installation
+------------
+
+I want everything
+~~~~~~~~~~~~~~~~~
+
```

Comment:
I think the migration page should cover the post install steps only as the installation page already has upgrade instructions and users will likely end up there first when running the upgrade.